io.sloeber.core/src/io/sloeber/core/api/BoardDescriptor.java

```diff
@@ -48,997 +50,1016 @@
-
-        this.myBoardsVariant = sourceBoardDescriptor.getBoardVariant();
-        this.myReferencedBoardVariantPlatformPath = sourceBoardDescriptor.getReferencedVariantPlatformPath();
-        this.myBoardsCore = sourceBoardDescriptor.getBoardsCore();
-        this.myReferencedCorePlatformPath = sourceBoardDescriptor.getReferencedCorePlatformPath();
-        this.myReferencedUploadToolPlatformPath = sourceBoardDescriptor.getReferencedUploadPlatformPath();
-        this.myUploadTool = sourceBoardDescriptor.getuploadTool();
-    }
-
-    public String getuploadTool() {
-        return this.myUploadTool;
-    }
-
-    private String getBoardsCore() {
-        return this.myBoardsCore;
-    }
-
-    /*
-	 * Sets default options as follows
-	 * If no option is specified take the first one
-     * if a option is specified but the value is invalid take the first one
-     *
-	 * this is so because I want to provide a list of options
-	 * but if the options are incomplete or invalid
-	 * this method still returns a complete and valid set.
-     */
-    private void setDefaultOptions() {
-
-        TreeMap<String, String> allMenuIDs = this.myTxtFile.getMenus();
-        for (Map.Entry<String, String> curMenuID : allMenuIDs.entrySet()) {
-            String providedMenuValue = this.myOptions.get(curMenuID.getKey());
-            ArrayList<String> menuOptions = this.myTxtFile.getMenuItemIDsFromMenuID(curMenuID.getKey(), getBoardID());
-            if (menuOptions.size() > 0) {
-              if (providedMenuValue == null) {
-        
-                    this.myOptions.put(curMenuID.getKey(), menuOptions.get(0));
-                }
-            else if (!menuOptions.contains(providedMenuValue)) {
-                    this.myOptions.put(curMenuID.getKey(), menuOptions.get(0));
-                }
-            }
-        }
-    }
-
-    /**
-     * tries to set the project to the boarddescriptor
-     *
-     * @param project
-     * @param monitor
-     * @return true if success false if failed
-     */
-    public boolean configureProject(IProject project, IProgressMonitor monitor) {
-        ICProjectDescription prjCDesc = CoreModel.getDefault().getProjectDescription(project);
-        ICConfigurationDescription configurationDescription = prjCDesc.getActiveConfiguration();
-        try {
-            save(configurationDescription);
-            // prjCDesc.setActiveConfiguration(configurationDescription);
-            CoreModel.getDefault().getProjectDescriptionManager().setProjectDescription(project, prjCDesc, true, null);
-        } catch (Exception e) {
-            e.printStackTrace();
-            Common.log(new Status(IStatus.ERROR, io.sloeber.core.Activator.getId(), "failed to save the board settings",
-                    e));
-            return false;
-        }
-        return true;
-    }
-
-    /*
-     * Method to create a project based on the board
-     */
-    public IProject createProject(String projectName, URI projectURI,
```

Comment:
This is the old create project that uses the "shouldhaveBeeninCDT.java" that has been deleted.